webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.js

```diff
@@ -0,0 +1,62 @@
+import React, { createContext } from 'react';
+import { Modal } from 'patternfly-react';
+import { omit } from 'lodash';
+import PropTypes from 'prop-types';
+import ForemanModalHeader from './subcomponents/ForemanModalHeader';
+import ForemanModalFooter from './subcomponents/ForemanModalFooter';
+
+export const ModalContext = createContext(null);
+
+const ForemanModal = props => {
+  const { isOpen, onClose, title } = props;
+  // Extract header and footer from children, if provided
+  let children = React.Children.toArray(props.children);
+  const headerChild = children.find(
+    child => child.type === ForemanModal.Header
+  );
+  const footerChild = children.find(
+    child => child.type === ForemanModal.Footer
+  );
+  children = children.filter(
+    child => child !== headerChild && child !== footerChild
```

Comment:
Can you reorganize it a function inorder  to improve reuse and code's readability?
```js
const findAChild = (children, type) => children.find(child => child.type === type);
```
Plus, we can create a function that combines `toArray` and `filter` together on children object:

```js
import { Children } from 'react';

Children.filter = (children, filterFn) => Children
    .toArray(children)
    .filter(filterFn);
};

export default Children;
```
```js
import Children from '../utils/children';
Children.filter(props.children, item => 'header' === item.type)
```

Reply:
@amirfefer I don't really understand why the other way would be better?

```javascript
children.filter(child => child !== headerChild && child !== footerChild)
```
It's just saying filter the children to any child where (a) it's not the Header, and (b) it's not the Footer.

If we extract that logic to a different file and two functions, and also make a new property on React.Children, it feels less readable to me.

If you're concerned with making the component too big, would it be acceptable if I just used the same logic extracted to a different file?

```javascript
  function extractHeaderAndFooterFromChildren(children) {
    // Extract header and footer from children, if provided
    let children = React.Children.toArray(props.children);
    const headerChild = children.find(
      child => child.type === ForemanModal.Header
    );
    const footerChild = children.find(
      child => child.type === ForemanModal.Footer
    );
    return children.filter(
      child => child !== headerChild && child !== footerChild
    );
  }
```

Or being as concise as possible: 

```javascript
function extractHeaderAndFooterFromChildren(children) {
    // Extract header and footer from children, if provided
    return React.Children.toArray(props.children).filter(
      child =>
        child.type !== ForemanModal.Header && child.type !== ForemanModal.Footer
    );
}
```

I would prefer that, or just leave it as is (unless I'm missing something..)

Reply:
> If you're concerned with making the component too big, would it be acceptable if I just used the same logic extracted to a different file?

I like that, it make sense of doing something like:
```js
const { header, footer, body } = extractModalNodes(children);
```

Reply:
@sharvit That's a better name, thanks :)  I will update it